### PR TITLE
improve Attribute format constraints

### PR DESCRIPTION
**Pull request by beetlecrunch**

### CODE REVIEW

**Comment by beetlecrunch** [comment]
I leave the maxLength set to 48.
Tell me your opinion.
In case I apply the same constraint to attribute-registry.

**Comment by galales** [comment]
not sure if I would remove the constraints entirely, what about expanding the regex instead?
And I agree, keeping the regex the same as the attribute registry is probably for the best 👍

**Comment by TonioGela** [comment]
> not sure if I would remove the constraints entirely, what about expanding the regex instead? And I agree, keeping the regex the same as the attribute registry is probably for the best 👍

I do agree. What about expanding `'[a-z0-9 \-]{,48}'` to `'[a-zA-Z0-9 \-]{,96}'` + some normalization code side (like uppercasing or lowercasing everything?)

**Comment by beetlecrunch** [comment]
> not sure if I would remove the constraints entirely, what about expanding the regex instead? And I agree, keeping the regex the same as the attribute registry is probably for the best 👍

Ok seems absolutely reasonable.

What about: `[a-zA-Z0-9 \-]{2,24}` ?

File: src/main/resources/interface-specification.yml [hunk 1]
```diff
@@ -1128,18 +1128,21 @@ components:
         code:
           type: string
           description: source identifier
-          pattern: '[a-z0-9 \-]{,48}'
-          maxLength: 48
+          pattern: '^[a-zA-Z0-9\-_]+$'
```
Comment (galales) [inline src/main/resources/interface-specification.yml:1131]: I see that in other regex the `_` is escaped, but not here.
Is this intentional?

Comment (TonioGela) [reply src/main/resources/interface-specification.yml:1131]: ☝️

Comment (beetlecrunch) [reply src/main/resources/interface-specification.yml:1131]: Removing the escape in the other patterns is not necessary.

Comment (TonioGela) [reply src/main/resources/interface-specification.yml:1131]: why?

Comment (beetlecrunch) [reply src/main/resources/interface-specification.yml:1131]: Is not necessary to escape "_".
Am I missing something?

Comment (TonioGela) [reply src/main/resources/interface-specification.yml:1131]: Hmm, tests with regex101.com using the ecmascript parser confirmed what you're saying

**Review by galales — APPROVED** [review]

**Review by TonioGela — APPROVED** [review]

**Pull request merged**